controllers/utils.js

```diff
@@ -33,39 +35,57 @@ function hasUserSignedUpForRelay(user) {
 
 function getExperimentBranch(req, sorterNum) {
 
+  if (req.session.excludeFromExperiment && !req.query.experimentBranch) {
```

Comment:
This is the primary fix for #1671. In summary, any time a session would return `false` for `getExperimentBranch()`, it would be forgotten when the function was called again (on another page load, refresh, etc). This applies specifically to a use who would be eligible to be in the experiment. They would have the experiment sorter run until it landed them in one of the experiment branches. 

One note about this if statement is that it includes an override for a query parameter. This is for testing purposes. If you add `?experimentBranch=vb`, it will place you in that cohort, regardless of getting discounted previously. One caveat is that if your language settings rule you out, this query request is denied (Again – for testing purposes).